io/include/traccc/io/read_geometry.hpp

```diff
@@ -27,4 +27,19 @@ namespace traccc::io {
 geometry read_geometry(std::string_view filename,
                        data_format format = data_format::csv);
 
+/// Read in the detector geometry description from a detector object
+template <typename detector_t>
+geometry alt_read_geometry(const detector_t& det) {
+
+    const auto transforms = det.transform_store();
+    const auto surfaces = det.surfaces();
+    std::map<traccc::geometry_id, traccc::transform3> maps;
```

Comment:
The ```traccc::geometry_id``` is just some hash right? Because the ```sf.barcode().value()``` is pretty much meaningless if not used in the barcode type. Using ```std::map<detray::barcode, traccc::transform3>``` would retain much more information, but I don't know how feasible it is at this point. I am guessing, that the ```geometry``` type is still something traccc specific that can't be changed?

Reply:
Barcode is used in downstream edms (`track_candidate` or `track_state` for KF).
I have not just fixed `traccc::geometry` yet and it is not an urgent issue as far as I concern (geometry information is barely used in clusterization and seeding)

Reply:
BTW, we are recording geometry id of hits as `sf.barcode().value()` in `detray::event_writer` - should this be fixed as well?

Reply:
But we are reading it back into a barcode or not? I am also not saying it will break, there is just no strict guarantee about the ordering of the values (Although I believe it still holds by design). It should be possible to sort the values and get all surfaces for one volume in a range, I think, but let's not rely on this.

I am afraid that someone might read a detray barcode into an ACTS geometry ID and get weird garbage out of it.

Reply:
I thought we can create barcode from geometry id. Could you check the following line?

https://github.com/acts-project/traccc/blob/3d8ddbc9182b12887d09d625e88b924ef17432a4/core/include/traccc/utils/seed_generator.hpp#L70

Reply:
There is also a similar line in #430 :

https://github.com/beomki-yeo/traccc/blob/d71faf68cb318d1bce8878728cb2092563d5e37d/core/src/seeding/track_params_estimation.cpp#L41

Reply:
You can transform a detray barcode to a 64bit unsigned integer type and back to a barcode, yes. But it also possible to instantiate a barcode on any other integer value, e.g. one that was encoded by ACTS. Then the detray barcode will decode garbage from that value. As long as ```surface_link``` was written out by detray (e.g. to a csv file), the above lines are OK. But if we use different barcodes/geometryID/source links in the same code and we accidentally mix them up, we might get into trouble

Reply:
Hence, it would be best to save the barcodes as ```detray:::barcode``` and not as an integer

Reply:
But how can I save the full information when simulation data only geometry ID (output of `barcode::value()`)?

Reply:
As I said, as long as the value originally came from a barcode its fine. But this values -> barcode conversion should only happen in the csv IO where we read it data from files that we know are generated by detray. I am more worried about this line https://github.com/acts-project/traccc/blob/3d8ddbc9182b12887d09d625e88b924ef17432a4/core/include/traccc/definitions/primitives.hpp#L26

Reply:
Sounds good